spec/util/app-service-mock.js

```diff
@@ -4,10 +4,52 @@ var util = require("util");
 var instance = null;
 
 function MockAppService() {
+    let self = this;
+
+    this.app = {
+        post: function(path, handler) {
+            if (path === '/_matrix/provision/link') {
+                self.link = handler;
+            }
+            else if (path === '/_matrix/provision/unlink') {
+                self.unlink = handler;
+            }
+            else {
+                throw new Error(`Unrecognised path for mock provisioning endpoint \"${path}\"`);
+            }
+        }
+    };
+
     EventEmitter.call(this);
 }
 util.inherits(MockAppService, EventEmitter);
 
+// Simulate a request to the link provisioning API
+//  parameters {object} - the API parameters
+//  statusCallback {function} - Called when the server returns a HTTP response code.
+//  jsonCallback {function} - Called when the server returns a JSON object.
+//  link {boolean} - true if this is a link request (false if unlink).
```

Comment:
It is more readable in tests if you can see the word "link" and "unlink", rather than `true` and `false` and having to remember which is which. I would just have `_link` and `_unlink` functions on the `MockAppService`.